File: oop/counter.py
class Counter:
    def __init__(self, n: int) -> None:
        self.n = n

    def increase(self) -> int:
        self.n += 1
        return self.n


counter = Counter(7)

counter.increase()

result = counter.increase()

print("Done, result= ", result)

File: oop/counter.cxx
#include <iostream>

using namespace std;

class Counter {
private:
  int n;

public:
  Counter(int ctor_n) : n(ctor_n) {}
  int increase() { return ++this->n; }
};

int main() {
  Counter *counter = new Counter(7);
  counter->increase();
  int result = counter->increase();
  cout << "Done, result= " << result << endl;
  delete counter;
}
